src/coreComponents/fieldSpecification/FieldSpecificationManager.cpp

```diff
@@ -105,41 +164,36 @@ void FieldSpecificationManager::validateBoundaryConditions( MeshLevel & mesh ) c
                             string const fieldName )
     {
       InputFlags const flag = fs.getWrapper< string >( FieldSpecificationBase::viewKeyStruct::fieldNameString() ).getInputFlag();
-
-      // 2.a) If we enter this loop, we know that the set has been created
-      //      Fracture/fault sets are created later and the "apply" call silently ignores them
       isTargetSetCreated.at( setName ) = 1;
-
-      // 2.b) If the fieldName is registered on this target, we record it
-      //      Unfortunately, we need two exceptions:
-      //       - FieldSpecification that do not target a field, like Aquifer, Traction, Equilibrium, etc. For these, the check is not
-      // necessary (the user cannot mess up)
-      //       - Face boundary conditions that target cell-based quantities, like the face BC of the flow solvers
-      if( targetGroup.hasWrapper( fieldName ) ||
-          flag == InputFlags::FALSE || // no need to check input if the input flag is false (Aquifer, Traction, Equilibrium do not target a
-                                       // field)
-          targetGroup.getName() == MeshLevel::groupStructKeys::faceManagerString() ) // the field names of the face BCs are not always
-                                                                                     // registered on
-                                                                                     // the faceManager...
+      if( targetGroup.hasWrapper( fieldName ) ||flag == InputFlags::FALSE ||
+          targetGroup.getName() == MeshLevel::groupStructKeys::faceManagerString() )     // the field names of the face BCs are not always
+                                                                                         // registered on
+                                                                                         // the faceManager...
```

Comment:
Why did you delete those comments? Not useful?
```suggestion
      InputFlags const flag = fs.getWrapper< string >( FieldSpecificationBase::viewKeyStruct::fieldNameString() ).getInputFlag();
      isTargetSetCreated.at( setName ) = 1;
      if( targetGroup.hasWrapper( fieldName ) ||flag == InputFlags::FALSE ||
          targetGroup.getName() == MeshLevel::groupStructKeys::faceManagerString() )     // the field names of the face BCs are not always
                                                                                         // registered on the faceManager...
```